frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1079,6 +1082,88 @@
+                            }
+                        }
+                        else if (duration._data.months !=0) {
+                            var months = duration.months();
+                            vm.completeLeaderboard[i].submission__submitted_at = months;
+                            if (months.toFixed(0)==1) {
+                                vm.completeLeaderboard[i].timeSpan = 'month';
+                            } else {
+                                vm.completeLeaderboard[i].timeSpan = 'months';
+                            }
+                        }
+                        else if (duration._data.days !=0) {
+                            var days = duration.asDays();
+                            vm.completeLeaderboard[i].submission__submitted_at = days;
+                            if (days.toFixed(0)==1) {
+                                vm.completeLeaderboard[i].timeSpan = 'day';
+                            } else {
+                                vm.completeLeaderboard[i].timeSpan = 'days';
+                            }
+                        }
+                        else if (duration._data.hours !=0) {
+                            var hours = duration.asHours();
+                            vm.completeLeaderboard[i].submission__submitted_at = hours;
+                            if (hours.toFixed(0)==1) {
+                                vm.completeLeaderboard[i].timeSpan = 'hour';
+                            } else {
+                                vm.completeLeaderboard[i].timeSpan = 'hours';
+                            }
+                        }
+                        else if (duration._data.minutes !=0) {
+                            var minutes = duration.asMinutes();
+                            vm.completeLeaderboard[i].submission__submitted_at = minutes;
+                            if (minutes.toFixed(0)==1) {
+                                vm.completeLeaderboard[i].timeSpan = 'minute';
+                            } else {
+                                vm.completeLeaderboard[i].timeSpan = 'minutes';
+                            }
+                        }
+                        else if (duration._data.seconds != 0) {
+                            var second = duration.asSeconds();
+                            vm.completeLeaderboard[i].submission__submitted_at = second;
+                            if (second.toFixed(0)==1) {
+                                vm.completeLeaderboard[i].timeSpan = 'second';
+                            } else {
+                                vm.completeLeaderboard[i].timeSpan = 'seconds';
+                            }
+                        }
+                    }
+                    console.log(vm.completeLeaderboard);
+                    vm.selectedPhaseSplit = response.data.results[0];
+                    vm.getCompleteLeaderboardTextOption = (vm.show_complete_leaderboard) ? "complete leaderboard":"get private submissions";
```

Comment:
Please fix the text to `Show all submissions` on UI.

Reply:
`Show all submissions` seems misleading here.

Can we instead change this to show: `Show private submissions` and `Hide private submissions` ?

Reply:
Hmmm..I'm thinking to keep the default the same as the public leaderboard and then if a host wants to see the private submissions then they can see.

Reply:
I am sorry but I didn't get your point. Can you please elaborate ?

Reply:
So, the default view for the host will be the same as that of the leaderboard visible to participants so that there isn't any inconsistency in ranks (we've received requests for it earlier regarding the inconsistency in the view of host and participant) and if the challenge host wants to see the hidden submissions then they can toggle using this button.

Feel free to ask if it is still not clear.

Reply:
Agree, makes sense. But I think it's better to show the label as `Include private submissions`  or `Exclude private submissions`

Reply:
yes, `Include private submissions` makes sense here.